R/map.R

```diff
@@ -160,3 +166,20 @@ walk <- function(.x, .f, ..., .progress = FALSE) {
   map(.x, .f, ..., .progress = .progress)
   invisible(.x)
 }
+
+with_indexed_errors <- function(expr, i, error_call = caller_env()) {
+  withCallingHandlers(
+    expr,
+    error = function(cnd) {
+      if (i == 0L) {
+        # Error happened before or after loop
+      } else {
+        cli::cli_abort(
+          "In index {i}.",
+          parent = cnd,
+          call = error_call
+        )
+      }
```

Comment:
```suggestion
      if (i != 0L) {
        # If `i == 0L`, error happened before or after loop
        cli::cli_abort(
          "In index {i}.",
          parent = cnd,
          call = error_call
        )
      }
```

Possibly cleaner, depending on how you feel about it

Reply:
I seem to be generally happy with an `if` branch that only contains a comment.